### PR TITLE
Bundling BPMN feature

### PR DESCRIPTION
**Pull request by Drifftr**

### CODE REVIEW

File: features/org.wso2.developerstudio.eclipse.bps.bpmn.feature/feature.common.properties [hunk 1]
```diff
@@ -0,0 +1,217 @@
+################################################################################
+# Copyright 2009 WSO2, Inc. (http://wso2.com)
```
Comment (hasithaa) [inline features/org.wso2.developerstudio.eclipse.bps.bpmn.feature/feature.common.properties]: Update year to 2016.

File: features/org.wso2.developerstudio.eclipse.bps.bpmn.feature/pom.xml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                                    </includes>
+                                </resource>
+                            </resources>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+</project>
```
Comment (hasithaa) [inline features/org.wso2.developerstudio.eclipse.bps.bpmn.feature/pom.xml]: End with a new Line.

File: plugins/org.wso2.developerstudio.bpmn.extensions/plugin.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?eclipse version="3.4"?>
+<plugin>
+<extension point="org.eclipse.ui.startup">
+</extension>
+ </plugin>
```
Comment (hasithaa) [inline plugins/org.wso2.developerstudio.bpmn.extensions/plugin.xml]: Format XML.

File: plugins/org.wso2.developerstudio.bpmn.extensions/pom.xml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+				<version>3.1</version>
+				<configuration>
+					<source>1.8</source>
+					<target>1.8</target>
+				</configuration>
+			</plugin>
+		</plugins>
+	</build>
+	<packaging>eclipse-plugin</packaging>
+</project>
```
Comment (hasithaa) [inline plugins/org.wso2.developerstudio.bpmn.extensions/pom.xml]: End with a new Line. Check this for all files.

File: plugins/org.wso2.developerstudio.bpmn.extensions/src/org/wso2/developerstudio/bpmn/extensions/Activator.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
+package org.wso2.developerstudio.bpmn.extensions;
```
Comment (hasithaa) [inline plugins/org.wso2.developerstudio.bpmn.extensions/src/org/wso2/developerstudio/bpmn/extensions/Activator.java]: Missing License header.

File: plugins/org.wso2.developerstudio.bpmn.extensions/src/org/wso2/developerstudio/bpmn/extensions/ExtensionConstants.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package org.wso2.developerstudio.bpmn.extensions;
+
+public class ExtensionConstants {
+
+    public static final String WSO2_TASK_LIB_LOCATION = "libs/org.wso2.developerstudio.bpmn.extensions-3.6.0-SNAPSHOT.jar";
+
```
Comment (hasithaa) [inline plugins/org.wso2.developerstudio.bpmn.extensions/src/org/wso2/developerstudio/bpmn/extensions/ExtensionConstants.java]: Don't hard code this value. We need to externalize this or pick dynamically at runtime. .

File: plugins/org.wso2.developerstudio.bpmn.extensions/src/org/wso2/developerstudio/bpmn/extensions/Activator.java [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+        return plugin;
+    }
+
+    /*
+     * Adds a user library to runtime eclipse as Activiti Designer Extension.
+     */
+    public void addUserLibrary() throws CoreException, URISyntaxException, IOException {
+        ClasspathContainerInitializer initializer = JavaCore
+                .getClasspathContainerInitializer(JavaCore.USER_LIBRARY_CONTAINER_ID);
+        String libraryName = ExtensionConstants.ACTIVITI_DESIGNER_EXTENSIONS;
```
Comment (hasithaa) [inline plugins/org.wso2.developerstudio.bpmn.extensions/src/org/wso2/developerstudio/bpmn/extensions/Activator.java]: String libraryName is redundant. use ExtensionConstants.ACTIVITI_DESIGNER_EXTENSIONS directly.

File: plugins/org.wso2.developerstudio.bpmn.extensions/src/org/wso2/developerstudio/bpmn/extensions/Activator.java [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+        Bundle bundle = Platform.getBundle(ExtensionConstants.BUNDLE_ID);
+        URL fileURL = bundle.getEntry(ExtensionConstants.WSO2_TASK_LIB_LOCATION);
+
+        File file = new File(FileLocator.resolve(fileURL).toURI());
+        IPath containerPath = new Path(JavaCore.USER_LIBRARY_CONTAINER_ID);
+        initializer.requestClasspathContainerUpdate(containerPath.append(libraryName), null, new IClasspathContainer() {
```
Comment (hasithaa) [inline plugins/org.wso2.developerstudio.bpmn.extensions/src/org/wso2/developerstudio/bpmn/extensions/Activator.java]: use ExtensionConstants.ACTIVITI_DESIGNER_EXTENSIONS directly.

File: plugins/org.wso2.developerstudio.bpmn.extensions/src/org/wso2/developerstudio/bpmn/extensions/Activator.java [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+            public IPath getPath() {
+                return new Path(JavaCore.USER_LIBRARY_CONTAINER_ID).append(libraryName);
```
Comment (hasithaa) [inline plugins/org.wso2.developerstudio.bpmn.extensions/src/org/wso2/developerstudio/bpmn/extensions/Activator.java]: use ExtensionConstants.ACTIVITI_DESIGNER_EXTENSIONS directly.

File: plugins/org.wso2.developerstudio.humantaskeditor/src/org/wso2/developerstudio/humantaskeditor/wizards/HumanTaskExportWizard.java [hunk 10]
```diff
@@ -69,6 +70,7 @@ public void run(IProgressMonitor monitor) throws InvocationTargetException {
     public void init(IWorkbench arg0, IStructuredSelection arg1) {
         this.selection = arg1;
         setWindowTitle(HumantaskEditorConstants.HUMAN_TASK_EXPORT_WIZARD_TITLE);
+        PlatformUI.getWorkbench().getHelpSystem().displayHelp("hteditor");
```
Comment (hasithaa) [inline plugins/org.wso2.developerstudio.humantaskeditor/src/org/wso2/developerstudio/humantaskeditor/wizards/HumanTaskExportWizard.java]: Move "hteditor" to a static final constants.

File: plugins/org.wso2.developerstudio.humantaskeditor/src/org/wso2/developerstudio/humantaskeditor/wizards/HumanTaskProjectWizard.java [hunk 11]
```diff
@@ -428,6 +428,8 @@ private static void addNature(IProject project) throws CoreException {
     @Override
     public void init(IWorkbench workbench, IStructuredSelection selection) {
         this.selection = selection;
+        PlatformUI.getWorkbench().getHelpSystem().displayHelp("hteditor");
```
Comment (hasithaa) [inline plugins/org.wso2.developerstudio.humantaskeditor/src/org/wso2/developerstudio/humantaskeditor/wizards/HumanTaskProjectWizard.java]: Move "hteditor" to a static final constants.

File: plugins/org.wso2.developerstudio.bpmn.extensions.activator/src/org/wso2/developerstudio/bpmn/extensions/StartupAction.java [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+        File libFolder = new File(new URI(url.getProtocol(), url.getPath(), null));
+        Files.walk(Paths.get(libFolder.getAbsolutePath())).forEach(
+                filePath -> {
+                    if (Files.isRegularFile(filePath)) {
+                        File file;
+                        try {
+                            file = new File(filePath.toUri());
+                            classPathEntries.add(JavaCore.newLibraryEntry(
+                                    new Path(file.getAbsolutePath()).makeAbsolute(), null, null));
+                        } catch (Exception e) {
```
Comment (nandika) [inline plugins/org.wso2.developerstudio.bpmn.extensions.activator/src/org/wso2/developerstudio/bpmn/extensions/StartupAction.java]: catch the specific exception.

**Pull request merged**